sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -848,8 +850,21 @@ def _delete_dataset(self, project_id, dataset_id, delete_contents=True):
   @retry.with_exponential_backoff(
       num_retries=MAX_RETRIES,
       retry_filter=retry.retry_on_server_errors_and_timeout_filter)
-  def create_temporary_dataset(self, project_id):
+  def create_temporary_dataset(self, project_id, source_table_reference=None):
     dataset_id = BigQueryWrapper.TEMP_DATASET + self._temporary_table_suffix
+    location = None
+
+    # Get location of source dataset
+    try:
+      tr = source_table_reference
+      if tr is not None:
+        if tr.projectId is None: table_project_id = project_id
```

Comment:
Nit: avoid inline statements and format the statements inside if on the next line.